node/src/actors/chain_manager/mod.rs

```diff
@@ -928,8 +928,22 @@ impl ChainManager {
         let current_epoch = if let Some(epoch) = self.current_epoch {
             epoch
         } else {
-            log::error!("Current epoch not loaded in ChainManager");
-            return;
+            let epoch = match self.chain_state.chain_info.as_ref() {
+                Some(chain_info) => {
+                    if block.hash() == chain_info.consensus_constants.genesis_hash {
+                        0
+                    } else {
+                        log::error!("Current epoch not loaded in ChainManager");
+                        return;
+                    }
+                }
+                None => {
+                    log::error!("Current epoch not loaded in ChainManager");
+                    return;
+                }
+            };
+
+            epoch
```

Comment:
Let's abuse the magic of Rust to avoid some code repetition:
```suggestion
            // If there is no epoch set, it's because the chain is yet to be bootstrapped, or because of a data race
            match self.chain_state.chain_info.as_ref() {
                // If the chain is yet to be bootstrapped (the block we are processing is the genesis block), set the epoch to zero
                Some(chain_info) if chain_info.consensus_constants.genesis_hash == block.hash() => {
                    0
                }
                // In case of data race, shortcut the function
                _ => {
                    log::error!("Current epoch not loaded in ChainManager");
                    return;
                }
            }
```